crates/ruff/src/message/gitlab.rs

```diff
@@ -100,11 +109,7 @@ impl Serialize for SerializedMessages<'_> {
 }
 
 /// Generate a unique fingerprint to identify a violation.
-fn fingerprint(
-    message: &Message,
-    start_location: &SourceLocation,
-    end_location: &SourceLocation,
-) -> String {
+fn fingerprint(message: &Message, salt: &str) -> String {
```

Comment:
I recommend changing fingerprint to return a `u64` instead of the serialized string to improve performance. Returning a u64 over string has the advantages:

* The set can store the u64 values directly without having to "clone" the heap-allocated strings (less memory requirement and fewer heap allocations)
* Hashing a string requires more CPU cycles than hashing a u64

```suggestion
fn fingerprint(message: &Message, salt: u64) -> u64 {
```

The serialization of the u64 to a hex encoded string can be moved to line 97 to only do it for the final fingerprint.